boards/posix/native_posix/doc/board.rst

```diff
@@ -361,11 +368,66 @@ The following peripherals are currently provided with this board:
   Normally the Zephyr application and HW models run in very little time
   on the host CPU, so this is a good enough approach.
 
-.. _NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME:
-   ../../../../reference/kconfig/CONFIG_NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME.html
-
 **Interrupt controller**:
   A simple yet generic interrupt controller is provided. It can nest interrupts
   and provides interrupt priorities. Interrupts can be individually masked or
   unmasked. SW interrupts are also supported.
 
+Shell support
+*************
+
+When the :ref:`Shell` subsystem is compiled with your application, the native
+standard input (`stdin`) will be redirected to the shell.
+You may use the shell interactively through the console,
+by piping another process output to it, or by feeding it a file.
+
+When using it interactively you may want to select the option
+:option:`CONFIG_NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME`.
+
+When feeding `stdin` from a pipe or file, the console driver will ensure
+reproducibility between runs of the process:
+
+- The execution of the process will be stalled while waiting for new `stdin`
+  data to be ready.
+
+- Commands will be fed to the shell as fast as the shell can process them.
+  To allow controlling the flow of commands to the shell, you may use the
+  driver directive ``!wait <ms>``.
+
+- When the file ends, or the pipe is closed the driver will stop attempting to
+  read it.
+
+Driver directives
+=================
+
+The console driver understands a set of special commands: driver directives.
+These directives are captured by the console driver itself and are not
+forwarded to the shell.
+These directives are:
+
+- ``!wait <ms>``: When received, the driver will pause feeding commands to the
```

Comment:
I would consider such things "dangerous".  User data is user data - why trick with it?

Reply:
These are "inband commands" meant to affect only the transport driver. They start with `!` so you can tell them apart from normal shell commands, and so that they will not conflict with future normal shell commands.
For the use case they are meant for (You have a file with all commands you want to send), it makes most sense to have them inline. For ex, a file with a content like:
```
noprompt
select kernel
uptime
!wait 500
uptime
!quit
```
(Imagine those commands are something more sensible, like for BLE commands)

Reply:
>for a good way to #ifdef all of that cod
All that code is ifdef'ed under NATIVE_POSIX_STDIN_CONSOLE

Reply:
Yeah, it's just not everyone knows that besides this RTOSesquely backward "console" which you implement here, Zephyr also has standard POSIXy console with getchar()/putchar() API. My concern was that implementation of that would conflict with the changes here (because they both call getchar()), but somehow it seems to work so far: https://github.com/zephyrproject-rtos/zephyr/pull/5947